ComponentKit/LayoutComponents/CKStackUnpositionedLayout.mm

```diff
@@ -15,10 +15,38 @@
 #import <ComponentKit/CKAssert.h>
 
 #import "ComponentUtilities.h"
+#import "CKComponentInternal.h"
 #import "CKComponentSubclass.h"
 #import "CKStackLayoutComponentUtilities.h"
 
 /**
+ 
+ */
+static CGFloat resolveCrossDimensionMaxForStretchChild(const CKStackLayoutComponentStyle &style,
+                                                       const CKStackLayoutComponentChild &child,
+                                                       const CGFloat stackMax,
+                                                       const CGFloat crossMax)
+{
+  // stretched children may have a cross direction max that is smaller than the minimum size constraint of the parent.
+  const CGFloat computedMax = (style.direction == CKStackLayoutDirectionVertical ?
+                               child.component.size.resolve(kCKComponentParentSizeUndefined).max.width :
+                               child.component.size.resolve(kCKComponentParentSizeUndefined).max.height);
+  return computedMax == INFINITY ? crossMax : computedMax;
+}
+
+static CGFloat resolveCrossDimensionMinForStretchChild(const CKStackLayoutComponentStyle &style,
+                                                    const CKStackLayoutComponentChild &child,
+                                                    const CGFloat stackMax,
+                                                    const CGFloat crossMin)
+{
+  // stretched children will have a cross dimension of at least crossMin, unless they explicitly define a child size
+  // that is smaller than the constraint of the parent.
+  return (style.direction == CKStackLayoutDirectionVertical ?
+             child.component.size.resolve(kCKComponentParentSizeUndefined).min.width :
+          child.component.size.resolve(kCKComponentParentSizeUndefined).min.height) ?: crossMin;
```

Comment:
Does the `?:` operator work? I thought it was specifically for `nil` values but perhaps it also works with any zero-like value?

Bit of a nit: the indentation feels a bit off here. Should line #45 be at the same indentation level as this line?